### PR TITLE
Notifications: Settings bottom sheet

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-3768

## What changed (plus any additional context for devs)

This PR converts the settings modal to a bottom sheet

### iOS

https://www.loom.com/share/5ae4d8e4c48f442b9e37c28cd5fdb569

### Android

https://www.loom.com/share/7e3b96776ef2480ea0176f424b8ecefa

### CODE REVIEW

File: src/screens/SettingsModal.js [hunk 1]
```diff
@@ -169,91 +162,87 @@ export default function SettingsModal() {
...
-                  key={key}
-                  name={key}
-                  options={{
-                    cardStyleInterpolator,
-                    title: getTitle(),
-                  }}
-                  title={getTitle()}
-                />
-              )
+          {() => (
```
Comment (osdnk) [inline src/screens/SettingsModal.js:183]: why is this a function?

**Comment by osdnk** [comment]
@jxom can you make a recording of what happens on smaller iPhones? Particularly, whether the dismissing interaction continues with a scrolling gesture if the content is scrollable.

File: src/screens/SettingsModal.js (hunk 1 above)
Comment (jxom) [reply src/screens/SettingsModal.js:183]: Oh this was already here. The only diff in this file is that I have removed the top level <Modal> component.

**Comment by jxom** [comment]
> @jxom can you make a recording of what happens on smaller iPhones? Particularly, whether the dismissing interaction continues with a scrolling gesture if the content is scrollable. 

@osdnk Yep can do when I'm back online!

**Comment by jinchung** [comment]
@jxom linting errors

**Comment by jxom** [comment]
@osdnk – doesn't look like the sheet dismisses on the scrollable content, however, this also didn't work in the modal view either... is there an easy way to hook these scroll views up with the sheet?

https://user-images.githubusercontent.com/7336481/175798784-f243c8c3-bef2-45f3-b182-25a3b297bb44.mp4

**Comment by terrysahaidak** [comment]
this is actually very annoying when you try scroll fast to the top and it dismisses.

**Review by BrodyHughes — CHANGES_REQUESTED** [review]
on android when you have a settings option open (backups, currency, etc) its impossible to swipe the settings sheet away (whereas it IS possible on ios).

**Comment by jxom** [comment]
Have made it so that when you swipe down on Android, it'll dismiss the sheet

**Comment by BrodyHughes** [comment]
Looks good to me! dm'd one more thing.

**Review by estebanmino — APPROVED** [review]
LGTM!
https://www.loom.com/share/6c69304779b142118e5c4a8e25cab46a

**Pull request merged**